internal/addon/addon_test.go

```diff
@@ -0,0 +1,100 @@
+package addon
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	v1 "open-cluster-management.io/api/work/v1"
```

Comment:
nit. Always prefix k8s apis with their package name.
```suggestion
	workv1 "open-cluster-management.io/api/work/v1"
```